### PR TITLE
UPSTREAM: 23213: provide best effort generation even if types are asy…

### PR DESCRIPTION
**Pull request by pweil-**
…mmetrical

https://github.com/kubernetes/kubernetes/pull/23213

### CODE REVIEW

File: pkg/api/v1beta3/conversion_generated.go [hunk 1]
```diff
@@ -6289,7 +6446,8 @@ func autoConvert_v1beta3_PodSpec_To_api_PodSpec(in *apiv1beta3.PodSpec, out *api
 	// in.HostIPC has no peer in out
 	// unable to generate simple pointer conversion for v1beta3.PodSecurityContext -> api.PodSecurityContext
 	if in.SecurityContext != nil {
-		if err := s.Convert(&in.SecurityContext, &out.SecurityContext, 0); err != nil {
+		out.SecurityContext = new(api.PodSecurityContext)
+		if err := apiv1beta3.Convert_v1beta3_PodSecurityContext_To_api_PodSecurityContext(in.SecurityContext, out.SecurityContext, s); err != nil {
```
Comment (pweil-) [inline pkg/api/v1beta3/conversion_generated.go]: Interesting, this was using reflection before which means the custom conversion wasn't even being called.

File: pkg/api/v1beta3/conversion_generated.go [hunk 2]
```diff
@@ -6692,6 +6866,7 @@ func init() {
 		autoConvert_api_ObjectReference_To_v1beta3_ObjectReference,
 		autoConvert_api_Parameter_To_v1beta3_Parameter,
 		autoConvert_api_PersistentVolumeClaimVolumeSource_To_v1beta3_PersistentVolumeClaimVolumeSource,
+		autoConvert_api_PodSecurityContext_To_v1beta3_PodSecurityContext,
```
Comment (pweil-) [inline pkg/api/v1beta3/conversion_generated.go]: also interesting, this is still generated and registered but not used.  Same thing occurs for the other custom conversion functions in v1beta3/conversion.go

**Comment by pweil-** [comment]
[test]

File: pkg/api/v1beta3/conversion_generated.go (hunk 1 above)
Comment (liggitt) [inline pkg/api/v1beta3/conversion_generated.go]: not true... reflection calls registered custom conversions

File: pkg/api/v1beta3/conversion_generated.go (hunk 2 above)
Comment (liggitt) [inline pkg/api/v1beta3/conversion_generated.go]: that's normal

File: pkg/api/v1beta3/conversion_generated.go (hunk 1 above)
Comment (pweil-) [inline pkg/api/v1beta3/conversion_generated.go]: TIL

File: pkg/api/v1beta3/conversion_generated.go (hunk 2 above)
Comment (pweil-) [inline pkg/api/v1beta3/conversion_generated.go]: ok, looked normal but wanted to call it out

**Comment by pweil-** [comment]
re[test]

**Comment by danmcp** [comment]
@pweil- Bump

**Comment by pweil-** [comment]
merged upstream but generation has been refactored since and we'll get the lastest in our rebase.  Closing

**Pull request closed without merging**